src/pages/studyView/charts/barChart/BarChart.tsx

```diff
@@ -48,6 +52,18 @@ const TILT_ANGLE = 50;
 export default class BarChart extends React.Component<IBarChartProps, {}> implements AbstractChart {
 
     private svgContainer: any;
+    
+    @observable
```

Comment:
whenever making an object observable, consider using @observable.ref, which makes the object reference observable, but not individual properties.  it's basically an immutable at that point.  there is a cost to making things deeply observable and we shouldn't pay it unless necessary.  this case is totally negligible, but good for illustration.  at 178, instead of setting x and separately, you can just set mousePosition to a new object with x,y properties.